django/core/checks/model_checks.py

```diff
@@ -64,8 +64,10 @@ def extract_operation(obj):
         """
         operation, args, keywords = obj, [], {}
         while hasattr(operation, 'func'):
-            args.extend(getattr(operation, 'args', []))
-            keywords.update(getattr(operation, 'keywords', {}))
+            # The or clauses here are redundant but work around a
+            # bug in functools.partial in certain Python versions.
```

Comment:
[The same bug was fixed in 3.5](https://mail.python.org/pipermail/python-checkins/2016-February/141214.html), so there may be circumstances where this can fail in Python 3 as well. Better leave it?